### PR TITLE
EL Compatible

### PR DESCRIPTION
**Pull request by roderickm**
Continuation of ajjahn/puppet-dns#41

### CODE REVIEW

File: spec/defines/dns__key_spec.rb [hunk 1]
```diff
@@ -12,9 +18,41 @@
...
+    ['rspec-key.key-header', 'rspec-key.key-secret', 'rspec-key.key-footer'].each do |fragment|
+        it { should contain_concat__fragment(fragment).with_ensure('present') }
+        it { should contain_concat__fragment(fragment).with_target('/etc/named/bind.keys.d/rspec-key.key') }
+        it { should contain_concat__fragment(fragment).with_require(['Exec[get-secret-from-rspec-key]', 'File[/etc/named/bind.keys.d/rspec-key.secret]']) }
+    end
+    it { should contain_concat__fragment('rspec-key.key-secret').with_source('/etc/named/bind.keys.d/rspec-key.secret') }
+  end
 end
+
+#On a RedHat OS
```
Comment (solarkennedy) [inline spec/defines/dns__key_spec.rb]: Go ahead and junk the commented out code... is it supposed to be here?

**Comment by solarkennedy** [comment]
Thank you for doing this! I know it is hard to revive dead code from abandoned PRs.
Can you address the one issue I have with the commented out tests and either remove them or fix them.

@ajjahn or @danzilio any objections/thoughts?

I think after this I'll push a conversion from `Modulefile` to `metadata.json` so we can encode that this this os is supported?

**Comment by roderickm** [comment]
You're welcome. Thanks for tolerating a less-than-polished contribution. Working on these is helping me better use git/github. 

I don't know for sure that the CentOS support is complete; I simply wrote tests for the el-compatible PR and merged master. From the comments left in #41, merging this may pave the way for @leonidlm to contribute his patches and review RHEL/CentOS compatibility.

**Comment by ajjahn** [comment]
@solarkennedy I don't have any objections; conversion to `metadata.json` sounds good too.

@roderickm @solarkennedy - Thanks guys!

**Pull request merged**